src/charm.py

```diff
@@ -94,6 +105,19 @@ def _configure_pebble(self) -> None:
             self._container.add_layer(self._container_name, layer, combine=True)
             self._container.restart(self._service_name)
 
+    def _fiveg_n4_information_is_correct(self) -> bool:
+        """The `fiveg_n4` relation is not mandatory and limited to 1.
```

Comment:
Why limited to 1? There could be multiple UPF's and in practice there will mostly be more than 1.

Reply:
That's the current implementation. The fiveg_n4 relation is limited to 1 on the metadata file. 
Yazan is already implementing the loop over multiple relations so this part of the code will change